bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/UILockListener.java

```diff
@@ -121,7 +121,8 @@ public UILockListener(Display display) {
 
 	@Override
 	public void aboutToRelease() {
-		if (isUI()) {
+		Thread uiThread = ui;
+		if (uiThread != null && uiThread == Thread.currentThread()) {
```

Comment:
For me it seems that `uiThread` would be better an `AtomicReference` and one could better perform `compareAndSet` (also at other places in this class) but that might be another change :-)